### ISSUE TITLE
Math module - Rename Bessel functions

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#19051]
The names for these functions lack outside context and are very short - I propose renaming them in the following way:

| old name | new name |
|-----|-----|
| j0 | firstBesselOrder0 |
| j1 | firstBesselOrder1 |
| jn | firstBesselOrderN |
| y0 | secondBesselOrder0 |
| y1 | secondBesselOrder1 |
| yn | secondBesselOrderN |

The documentation should use the old names, in case a user is searching for them based on that name.

**Comment by bradcray** [comment]
My standard question:  What do other modern Math libraries (for languages we care about) call these?  If the short names are what everyone expects, I'd keep them (and having them be in a non-auto-used module makes that seem less bad).  If modern convention has gone to other/longer names, that sounds good to me as well.

**Comment by lydia-duncan** [comment]
Python's SciPy library uses these names, but puts them in a special module (literally scipy.special).

C uses those names, and gcc includes them by default, which caused a user on stackoverflow to run into problems when they tried to define a variable with the same name (more a problem for the y variants than the j variants, as y is frequently used in other scenarios).

Rust puts its Bessel functions in their own library but also uses a variation of the same name (Y0 rather than y0 but still j0, maybe due to y0 being something users would type regularly?)

I can go look at other languages, but it seems pretty unified thus far

**Comment by bradcray** [comment]
Yeah, this suggests to me that we should probably stick with the current names, and that making them not be auto-available would reduce chances of confusion.  We wouldn't have the C/gcc problem because of not having a global flat namespace, so the worst case here (I think) would be a user who `use`s rather than `import`s `Math`; thinks they've shadowed one of these routines, but has failed to for some reason; and then tries to refer to their shadow variables but gets these instead.

**Comment by lydia-duncan** [comment]
Do you think it would be worth putting them in their own special submodule as well?  It'd only be the six of them so far as I know, but 6 isn't too few and it did seem like there were other ones that could be added later if someone missed them.

**Comment by bradcray** [comment]
I probably wouldn't personally, unless we did that for more Math routines in general or saw an inherent problem with keeping them in Math (I don't really, particularly in a non-auto-use world).

**Comment by lydia-duncan** [comment]
I think we're not going to do this, based on the discussion, so I'm just going to close the issue

**Comment by lydia-duncan** [comment]
On the tgamma/lgamma issue (#19022), Damian indicated that he'd like us to think a bit more about the names of these functions and potentially their location. Reopening, but removing the 2.0 label as we're planning on marking them unstable for 2.0 and discussing after that (given how soon the 2.0 release candidate is approaching)

**Comment by lydia-duncan** [comment]
Clearing out old notes, I found the start of some investigation I did for doing a design discussion on them.  It was not completed and we prioritized other things, but I figured it would help us when we get back to it to have recorded somewhere more public:

Bessel prep

1. What is the API being presented?

```chapel
  /* Returns the Bessel function of the first kind of order `0` of `x`. */
  inline proc j0(x: real(32)): real(32) { … }
  inline proc j0(x: real(64)): real(64) { … }

  /* Returns the Bessel function of the first kind of order `1` of `x`. */
  inline proc j1(x: real(32)): real(32) { … }
  inline proc j1(x: real(64)): real(64) { … }

  /* Returns the Bessel function of the first kind of order `n` of `x`. */
  inline proc jn(n: int, x: real(32)): real(32) { … }
  inline proc jn(n: int, x: real(64)): real(64) { … }

  /* Returns the Bessel function of the second kind of order `0` of `x`, where
     `x` must be greater than 0 */
  inline proc y0(x: real(32)): real(32) { … }
  inline proc y0(x: real(64)): real(64) { … }

  /* Returns the Bessel function of the second kind of order `1` of `x`,
     where `x` must be greater than 0 */
  inline proc y1(x: real(32)): real(32) { … }
  inline proc y1(x: real(64)): real(64) { … }

  /* Returns the Bessel function of the second kind of order `n` of `x`,
     where `x` must be greater than 0 */
  inline proc yn(n: int, x: real(32)): real(32) { … }
  inline proc yn(n: int, x: real(64)): real(64) { … }
```

The names of these functions do not indicate what they do, though that can be determined by reading their documentation.  The `y` variants are somewhat likely to conflict with user symbols, but this is less of an issue now that they are no longer included in all user programs by default.

- How is it intended to be used?

Wikipedia says they are a generalization of the sine function (https://en.wikipedia.org/wiki/Bessel_function).  

- How is it being used in Arkouda and CHAMPS?

TODO: in arkouda

2. What's the history of the feature, if it already exists?

We recently decided to stop including these functions in all user programs by default. 

3. What's the precedent in other languages, if they support it?

a. Python
b. C/C++
c. Rust
d. Swift
e. Julia
f. Go 

4. Are there known Github issues with the feature? 
5. Are there features it is related to? What impact would changing or adding this feature have on those other features? 
6. How do you propose to solve the problem?

**Comment by lydia-duncan** [comment]
[On the erf issue](https://github.com/chapel-lang/chapel/issues/19013#issuecomment-3617497929), I recorded that Damian pointed out discussion is currently happening around many math functions in C.  I'm not sure if the Bessel functions are included in that, but the recommendation to make sure we take an up-to-date look at their state in C when we do stabilize these functions is likely still a good one (especially given that my notes here were from 2022 and I did not take action to update them for any changes since then)
